usr/lib/xtable/libxtable.c

```diff
@@ -0,0 +1,182 @@
+/*
+* SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <lauxlib.h>
+#include <lua.h>
+#include <lualib.h>
+
+#include <xtables.h>
+
+#include "luaxtable.h"
+
+#ifndef LUAXTABLE_MODULE
+#error "LUAXTABLE_MODULE is not defined"
+#endif
+
+static lua_State *L = NULL;
+const char *libxtable_match_key = "libxtable_match";
+const char *libxtable_target_key = "libxtable_target";
+
+int luaopen_libxtable(lua_State *L);
+
+static inline int libxtable_run(lua_State *L, const char *func_name, const char *key, int nargs, int nresults)
+{
+	int ret = 0;
+	int base = lua_gettop(L) - nargs;
+	lua_rawgetp(L, LUA_REGISTRYINDEX, key);
+
+	if (lua_getfield(L, -1, func_name) != LUA_TFUNCTION) {
+		fprintf(stderr, "Function %s not found\n", func_name);
```

Comment:
I would define a macro for `fprint(stderr, ...)` anyway.. to avoid repeating it all around the code..